bamboo/integration_tests/test_integration_lenet.py

```diff
@@ -0,0 +1,204 @@
+expected_mini_batch_times = {
+    'pascal':   0.0040,
+    'catalyst': 0.0055,
+    'lassen':   0.0020,
+    'ray':      -1,
+    'corona':   -1,
+}
+
+# ==============================================
+# Setup LBANN experiment
+# ==============================================
+
+def setup_experiment(lbann):
+    """Construct LBANN experiment.
+
+    Args:
+        lbann (module): Module for LBANN Python frontend
+
+    """
+    trainer = lbann.Trainer()
+    model = construct_model(lbann)
+    data_reader = construct_data_reader(lbann)
+    optimizer = lbann.SGD(learn_rate=0.01, momentum=0.9)
+    return trainer, model, data_reader, optimizer
+
+def construct_model(lbann):
+    """Construct LBANN model.
+
+    Args:
+        lbann (module): Module for LBANN Python frontend
+
+    """
+
+    # TODO (tym): Figure out how to switch between LBANN builds. See
+    # GitHub Issue #1289.
+    import lbann.models
+
+    # Layer graph
+    input_ = lbann.Input()
+    images = lbann.Identity(input_)
+    labels = lbann.Identity(input_)
+    x = lbann.models.LeNet(10)(images)
+    probs = lbann.Softmax(x)
+    loss = lbann.CrossEntropy([probs, labels])
+    acc = lbann.CategoricalAccuracy([probs, labels])
+
+    # Objects for LBANN model
+    callbacks = [lbann.CallbackPrint(), lbann.CallbackTimer()]
+    metrics = [lbann.Metric(acc, name='accuracy', unit='%')]
+
+    # Construct model
+    return lbann.Model(mini_batch_size,
+                       num_epochs,
+                       layers=lbann.traverse_layer_graph(input_),
+                       objective_function=loss,
+                       metrics=metrics,
+                       callbacks=callbacks)
+
+def construct_data_reader(lbann):
+    """Construct Protobuf message for Python data reader.
+
+    The Python data reader will import the current Python file to
+    access the sample access functions.
+
+    Args:
+        lbann (module): Module for LBANN Python frontend
+
+    """
+
+    # TODO (tym): Figure out how to switch between LBANN builds. See
+    # GitHub Issue #1289.
+    import lbann.contrib.lc.paths
+
+    # Load data readers from prototext
+    dirname = os.path.dirname
+    lbann_dir = dirname(dirname(dirname(os.path.realpath(__file__))))
+    pb_file = os.path.join(lbann_dir,
+                           'model_zoo',
+                           'data_readers',
+                           'data_reader_mnist.prototext')
+    message = lbann.lbann_pb2.LbannPB()
+    with open(pb_file, 'r') as f:
+        google.protobuf.text_format.Merge(f.read(), message)
+    message = message.data_reader
+
+    # Set location of MNIST data
+    for reader in message.reader:
+        reader.data_filedir = lbann.contrib.lc.paths.mnist_dir()
+
+    # No validation set
+    message.reader[0].validation_percent = 0
+
+    return message
+
+# ==============================================
+# Setup PyTest
+# ==============================================
+
+def augment_test_func(test_func):
```

Comment:
This is identical to the alexnet `augment_test_func` except for not skipping Nightly/CPU builds (and also not including `top-5` on the accuracy). Seems like it would be better to pull this out into a common function. The late-binding closures wouldn't be affected by that, I believe.

Reply:
At a certain point, I don't think it's worthwhile reducing code duplication if it makes the code harder to understand or if it constrains us to a rigid framework. I was thinking about an alternate approach where we pass callbacks into the `create_tests` function to perform the various assertions. It's fairly general and it allows us to consolidate some code, but I suspect it's hard to understand.

Reply:
I think reducing code duplication makes the code easier to understand - but we could argue this _ad infinitum_. I would have to see the callback approach to see how clean that is, but that seems like another good option.

In any case, if the code runs and the tests pass, then the PR is good to merge. Perhaps LBANN should at some point consider discussing code style guidelines (including things like "code duplication v factoring out common code").